File: cypher_b64.py
#!/usr/bin/env python3

from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import base64

input = '80:c9:55:64:73:f0'
CRYPTO_KEY= '8ookgvdIiH2YOgBnAju6Nmxtp14fn8d3'
CRYPTO_IV= 'rBEssDfxofOveRxR'

BLOCK_SIZE = 16


def Encrypt(data, key, iv):
    key = bytes(key, 'utf-8')
    iv = bytes(iv, 'utf-8')

    aes = AES.new(key, AES.MODE_CBC, iv)
    encrypted = aes.encrypt(pad(data.encode(), BLOCK_SIZE))
    
    # Make sure to strip "=" padding since urlsafe-base64 node module strips "=" as well
    return base64.urlsafe_b64encode(encrypted).decode("utf-8").rstrip("=")


def Decrypt(edata, key, iv):
    key = bytes(key, 'utf-8')
    iv = bytes(iv, 'utf-8')

    # Add "=" padding back before decoding
    edata = base64.urlsafe_b64decode(edata + '=' * (-len(edata) % 4))
    aes = AES.new(key, AES.MODE_CBC, iv)
    return unpad(aes.decrypt(edata), BLOCK_SIZE).decode("utf-8")


def main():
    output = Encrypt(input, CRYPTO_KEY, CRYPTO_IV)
    plaintext = Decrypt(output, CRYPTO_KEY, CRYPTO_IV)

    print(output)
    print(plaintext)


if __name__ == "__main__":
    main()

File: cypher_b64.js
#!/usr/bin/env node

const crypto = require('crypto');
const URLSafeBase64 = require('urlsafe-base64');

const input = "80:c9:55:64:73:f0";
const CRYPTO_KEY = "8ookgvdIiH2YOgBnAju6Nmxtp14fn8d3";
const CRYPTO_IV = "rBEssDfxofOveRxR";

// async function encrypt(val) {
function encrypt(val) {
  let cipher = crypto.createCipheriv('aes-256-cbc', CRYPTO_KEY, CRYPTO_IV);
  // UTF-8 to Base64 encoding
  let encrypted = cipher.update(val, 'utf8', 'base64');
  encrypted += cipher.final('base64');
  return URLSafeBase64.encode(encrypted);
};

// async function decrypt(encrypted) {
function decrypt(encrypted) {
  encrypted = URLSafeBase64.decode(encrypted);
  let decipher = crypto.createDecipheriv('aes-256-cbc', CRYPTO_KEY, CRYPTO_IV);
  // Base64 to UTF-8 decoding
  let decrypted = decipher.update(encrypted, 'base64', 'utf8');
  return (decrypted + decipher.final('utf8'));
};

var encrypted_key = encrypt(input);
var original_input = decrypt(encrypted_key);

console.log(encrypted_key);
console.log(original_input);
